### PR TITLE
Ch1148/capsvc call kafka janitor s api for topic

### PR DESCRIPTION
**Pull request by SEQUOIIA**

### CODE REVIEW

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |7|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 126 lines, deletes 24 lines of code in 9 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamemd. Skipping code health checks for those files.</li></ul>|

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |8|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 219 lines, deletes 55 lines of code in 14 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>5 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamemd. Skipping code health checks for those files.</li></ul>|

File: src/CapabilityService.WebApi/Features/Kafka/Domain/Events/TopicCreated.cs [hunk 1]
```diff
@@ -14,9 +15,30 @@ public TopicCreated(Topic topic)
 			Partitions = topic.Partitions;
 		} 
 		
+		[JsonConverter(typeof(TopicNameConverter))]
```
Comment (kim-lindhard-dfds) [inline src/CapabilityService.WebApi/Features/Kafka/Domain/Events/TopicCreated.cs]: Our domain should never concern itself with transport, what is the issue we are solving with the converter?

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |8|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 219 lines, deletes 96 lines of code in 14 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>5 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamemd. Skipping code health checks for those files.</li></ul>|

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |8|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 263 lines, deletes 114 lines of code in 15 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>6 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamemd. Skipping code health checks for those files.</li></ul>|

File: src/CapabilityService.WebApi/Infrastructure/Persistence/CapabilityServiceDbContext.cs [hunk 2]
```diff
@@ -1,73 +1,76 @@
...
-            modelBuilder.Entity<MessageContract>(cfg =>
-            {
-                cfg.ToTable("MessageContract");
-                cfg.HasKey(x => x.Type);
-            });
-        }
-    }
-}
\ No newline at end of file
+﻿using DFDS.CapabilityService.WebApi.Domain.Models;
```
Comment (kim-lindhard-dfds) [inline src/CapabilityService.WebApi/Infrastructure/Persistence/CapabilityServiceDbContext.cs]: Seems like line endlings

**Review by kim-lindhard-dfds — CHANGES_REQUESTED** [review]

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |8|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 263 lines, deletes 114 lines of code in 15 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>6 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamemd. Skipping code health checks for those files.</li></ul>|

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |8|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 269 lines, deletes 134 lines of code in 16 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>6 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamemd. Skipping code health checks for those files.</li></ul>|

File: src/CapabilityService.WebApi/Features/Kafka/Domain/Events/TopicCreated.cs (hunk 1 above)
Comment (SEQUOIIA) [reply src/CapabilityService.WebApi/Features/Kafka/Domain/Events/TopicCreated.cs]: It solves the issue of 

{
  "Name": {
      "name": "here"
    },
    "Description": "This is a thing",
    ....
}

File: src/CapabilityService.WebApi/Infrastructure/Persistence/CapabilityServiceDbContext.cs (hunk 2 above)
Comment (SEQUOIIA) [reply src/CapabilityService.WebApi/Infrastructure/Persistence/CapabilityServiceDbContext.cs]: Yes.

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |8|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 271 lines, deletes 158 lines of code in 16 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>6 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamemd. Skipping code health checks for those files.</li></ul>|

**Review by kim-lindhard-dfds — APPROVED** [review]

**Pull request merged**